src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx

```diff
@@ -59,18 +63,32 @@ export function OpenSavedQueryFlyout({
   const [languageFilterOptions, setLanguageFilterOptions] = useState<string[]>([]);
   const [selectedQuery, setSelectedQuery] = useState<SavedQuery | undefined>(undefined);
   const [searchQuery, setSearchQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
+  const [isLoading, setIsLoading] = useState(false);
+  const currentTabIdRef = useRef(selectedTabId);
 
   const fetchAllSavedQueriesForSelectedTab = useCallback(async () => {
-    const allQueries = await savedQueryService.getAllSavedQueries();
-    const templateQueriesPresent = allQueries.some((q) => q.attributes.isTemplate);
-    const queriesForSelectedTab = allQueries.filter(
-      (q) =>
-        (selectedTabId === 'mutable-saved-queries' && !q.attributes.isTemplate) ||
-        (selectedTabId === 'template-saved-queries' && q.attributes.isTemplate)
-    );
-    setSavedQueries(queriesForSelectedTab);
-    setHasTemplateQueries(templateQueriesPresent);
-  }, [savedQueryService, selectedTabId, setSavedQueries]);
+    setIsLoading(true);
+    if (queryStringManager.getQuery()?.dataset?.type === 'SECURITY_LAKE') {
+      setHasTemplateQueries(true);
+    }
+    if (currentTabIdRef.current === 'mutable-saved-queries') {
+      const allQueries = await savedQueryService.getAllSavedQueries();
+      const mutableSavedQueries = allQueries.filter((q) => !q.attributes.isTemplate);
+      if (currentTabIdRef.current === 'mutable-saved-queries') {
+        setSavedQueries(mutableSavedQueries);
+      }
+    } else if (currentTabIdRef.current === 'template-saved-queries') {
+      const query = queryStringManager.getQuery();
+      if (query?.dataset?.type) {
+        const templateQueries = await queryStringManager
+          .getDatasetService()
+          ?.getType(query.dataset.type)
+          ?.getSampleQueries?.();
+        if (Array.isArray(templateQueries)) setSavedQueries(templateQueries);
+      }
+    }
+    setIsLoading(false);
+  }, [savedQueryService, currentTabIdRef, setSavedQueries, queryStringManager]);
```

Comment:
Nit: we don't need to add `currentTabIdRef` and `setSavedQueries` in the dependency list. They are stable in same component.